src/kernels/hip_float8.hpp

```diff
@@ -478,19 +478,26 @@ inline MIOPEN_HIP_HOST_DEVICE bool operator>(const miopen_f8::hip_f8<T>& lhs,
 template <miopen_f8::hip_f8_type T>
 inline MIOPEN_HIP_HOST_DEVICE miopen_f8::hip_f8<T> fabs(miopen_f8::hip_f8<T> v)
 {
+#if !MIOPEN_FP8_IEEE_EXPONENT_BIAS
+    // Preserve "universal" nan.
+    // Otherwise it will be converted to valid 0.
+    if(v.data == 0x80)
+        return v;
+#endif
     v.data = v.data & 0x7f;
     return v;
 }
+
 template <class T>
-MIOPEN_HIP_HOST_DEVICE T F8_Max()
+MIOPEN_HIP_HOST_DEVICE T Generate(uint8_t bits)
 {
     union
     {
         uint8_t bits;
         T value;
     } x;
 
-    x.bits = 0x7F;
+    x.bits = bits;
     return x.value;
 }
```

Comment:
Formally speaking it is UB in `C++` (but allowed and perfectly fine in `C`).
Until C++20 `std::bitcast`, the only true way to do type punning is `std::memcpy` or any memcpy builtin.

Reply:
Thought it works.

Reply:
@CAHEK7 Behavior of this is unspecified in the Standard. But AFAICS this code came from the HIP library, and proven to work for us (we can consider this as documented). 

But this is a bit out of scope of this PR; I am only fixing the values.

Reply:
It is not `unspecified` it is clearly 'undefined' in the Standart. The difference between C and C++ is that in C++ union may have members with non-trivial constructors, destructors and so on. As I've said - it works, but still.

I know about that HIP library flaw, it's not a reference, it's a library which uses some highly-likely working the trick.
Btw, nvidia has it's own builtin for that sort of casting (I've checked  two implementations of `__shlf_xor` builtins).

https://en.cppreference.com/w/cpp/language/union
I know it's not an official version of the Standard, but it's quite close to be it.

And I'm not asking to change this code right now, I'm just pointing to that common delusion.

Reply:
@CAHEK7 
> It is not unspecified it is clearly undefined

I see. And that means that our program is ill-formed from the POV of the Standard.

However, in our implementation of the HIP compiler, it is well-defined so far.

I think this should be discussed with HIP compiler/library developers, as proposed in the PR description.

Reply:
@atamazov the runtime folks claims no changes were made at least recently to the above mentioned places, and the header files are located at:
https://github.com/ROCm/HIP/blob/355b16b473090b219a6a076cab902b6497ed1883/include/hip/hip_bfloat16.h
https://github.com/ROCm/clr/blob/develop/hipamd/include/hip/amd_detail/amd_hip_bfloat16.h

Reply:
@junliume Thanks. I do not see any issues in https://github.com/ROCm/clr/blob/develop/hipamd/include/hip/amd_detail/amd_hip_bfloat16.h. Although they don't define things from `<limits>` like our hip_bfloat16.h does, so maybe those things are a recent addition we made (and those issues are fixed in this PR).

But what about FP8/BF8 related things? Or header is a copy of
https://github.com/ROCm/MIOpen/blob/3a33fe7a1a2d2d30c6b0d6fe64efbca94d69ba2a/src/kernels/hip_float8.hpp#L34
and I've made a lot of fixes in `<limits>` and `<cmath>` things. Maybe these things are a recent addition to the original code, but I would like to make sure that we and compiler guys are on the same page about what F8 is.

Reply:
@JehandadKhan could you chip in on the F8 header alignment?